apollo-router/src/plugins/authentication.rs

```diff
@@ -353,37 +358,41 @@ impl Plugin for AuthenticationPlugin {
                         }
                     };
 
-                    request.context.enter_active_request().await;
-                    // Get the JWKS here
-                    let jwks_opt = match my_jwks.get(my_jwks_url).await {
-                        Ok(k) => k,
-                        Err(e) => {
-                            request.context.leave_active_request().await;
+                    // Search our set of JWKS to find the kid and process it
+                    // Note: This will search through JWKS in the order in which they are defined
+                    // in configuration. If the same "kid" is present in multiple JWKS, the router
+                    // will just use the first "kid" encountered.
+                    for jwks_url in my_jwks_urls {
+                        request.context.enter_active_request().await;
+                        // Get the JWKS here
+                        let jwks_opt = match my_jwks.get(jwks_url).await {
```

Comment:
Is there a way or is it possible to cache this ? Maybe it's a dumb question, I'm just asking in order to not call this everytime.
Also, if we're processing other URLs, something I think could be concerning is that the only right jwks is always at the end of the array, then we are processing all the vec each time. I don't know if it's an issue but if we can find a quick solution maybe that would be better for perfs

Reply:
my_jwks is a deduplicating cache, so most of the time this will just be a search through cached JWKS.

I agree that searching the vec of cached results is not ideal. However for most users it should be a single IDP or a small number and so performance is probably fine.

Reply:
Thanks